tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -105,6 +199,70 @@ static void test_advx_main(void)
 
 	k_sleep(K_MSEC(1000));
 
+	printk("Create advertising set...");
+	err = bt_le_ext_adv_create(BT_LE_ADV_CONN_NAME, &adv_callbacks, &adv);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	printk("Start advertising using extended commands (max_events)...");
+	is_sent = false;
+	ext_adv_param.timeout = 0;
+	ext_adv_param.num_events = 3;
+	err = bt_le_ext_adv_start(adv, &ext_adv_param);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	printk("Waiting...");
+	while (is_sent) {
```

Comment:
is_sent is initialised to false here, and set to true in the sent_cb. As a result the sleep loop in line 221 is never executed (if it was it would be a forever-loop, never continuing). 
If we want to make sure that advertisement has started it should be **while (!is_sent)** and a timeout mechanism added
If the intention is to wait here until the advertising is terminated we need to set is_sent to false in the appropriate place, and we still need to add the code to ensure that advertisement has started

Reply:
Fixed

Reply:
Ok, note however this test still assumes that the function is working correct, otherwise you'll have an endless loop. I.e. it is good enough for debugging, but not for an automated test like C.I.